packages/dds/tree/src/core/schema-stored/schema.ts

```diff
@@ -136,46 +118,166 @@ export const storedEmptyFieldSchema: TreeFieldStoredSchema = {
 	types: new Set(),
 };
 
+/**
+ * Opaque type erased handle to the encoded representation of the contents of a stored schema.
+ * @alpha
+ */
+export interface ErasedTreeNodeSchemaDataFormat extends Erased<"TreeNodeSchemaDataFormat"> {}
+export interface BrandedTreeNodeSchemaDataFormat
+	extends Brand<TreeNodeSchemaDataFormat, ErasedTreeNodeSchemaDataFormat> {}
+
+/**
+ * @alpha
+ */
+export abstract class TreeNodeStoredSchema {
+	protected _typeCheck!: MakeNominal;
+
+	/**
+	 * @privateRemarks
+	 * Returns TreeNodeSchemaDataFormat.
+	 * This is uses an opaque type to avoid leaking these types out of the package,
+	 * and is runtime validated by the codec.
+	 */
+	public abstract encode(): ErasedTreeNodeSchemaDataFormat;
+}
+
 /**
  * @alpha
  */
-export interface TreeNodeStoredSchema {
+export class ObjectNodeStoredSchema extends TreeNodeStoredSchema {
 	/**
+	 * @param objectNodeFields -
 	 * Schema for fields with keys scoped to this TreeNodeStoredSchema.
-	 *
 	 * This refers to the TreeFieldStoredSchema directly
 	 * (as opposed to just supporting FieldSchemaIdentifier and having a central FieldKey -\> TreeFieldStoredSchema map).
 	 * This allows os short friendly field keys which can ergonomically used as field names in code.
 	 * It also interoperates well with mapFields being used as a map with arbitrary data as keys.
 	 */
-	readonly objectNodeFields: ReadonlyMap<FieldKey, TreeFieldStoredSchema>;
+	public constructor(
+		public readonly objectNodeFields: ReadonlyMap<FieldKey, TreeFieldStoredSchema>,
+	) {
+		super();
+	}
 
+	public override encode(): ErasedTreeNodeSchemaDataFormat {
+		return brandErased<BrandedTreeNodeSchemaDataFormat>({
+			object: [...this.objectNodeFields]
+				.map(([k, v]) => encodeNamedField(k, v))
+				.sort(compareNamed),
```

Comment:
is sorting specifically part of the persisted format? can we document that if so?

Reply:
sorting is being done to ensure determinism. I have added a comment about it.